src/slic3r/GUI/Plater.cpp

```diff
@@ -8107,8 +8107,35 @@ void Plater::calib_pa(const Calib_Params& params) {
     wxGetApp().mainframe->select_tab(size_t(MainFrame::tp3DEditor));
     if (params.mode == CalibMode::Calib_PA_Line) {
         add_model(false, Slic3r::resources_dir() + "/calib/PressureAdvance/pressure_advance_test.stl");
-    }
-    else {
+
+        auto obj = model().objects[0];
+        auto printer_config = &wxGetApp().preset_bundle->printers.get_edited_preset().config;
+        auto bed_shape = printer_config->option<ConfigOptionPoints>("printable_area")->values;
+        BoundingBoxf bed_ext = get_extents(bed_shape);
+
+        // TODO: How to get the 3.5 from `calib_pressure_advance::calib_pressure_advance` instead of hardcoding
+        auto pa_calib_y_line_spacing = 3.5;
+        auto pa_calib_y_margin = 20.0;
+        auto pa_calib_total_steps = std::ceil((params.end - params.start) / params.step) + 1;
+        // The ideal sizes are the sizes that would fit the object without scaling. X is hardcoded to 120mm
+        // based on `calib_pressure_advance::generate_test`
+        auto pa_calib_y_ideal_size = pa_calib_total_steps * pa_calib_y_line_spacing + pa_calib_y_margin * 2;
+        auto pa_calib_x_ideal_size = 120.0;
+        // The real size is the ideal size if possible or the maximum size that fits the bed (with some margin).
+        // This is safe since the test generator clamps it to the bed size anyway.
+        auto pa_calib_y_real_size = std::min((bed_ext.size().y() - 10), pa_calib_y_ideal_size);
+        auto pa_calib_x_real_size = std::min((bed_ext.size().x() - 10), pa_calib_x_ideal_size);
+
+        obj->scale(
+            pa_calib_x_real_size / obj->bounding_box().size().x(), 
+            pa_calib_y_real_size / obj->bounding_box().size().y(), 
+            1.0
+        );
+
+        changed_objects({ 0 });
+        wxGetApp().get_tab(Preset::TYPE_PRINT)->update_dirty();
+        wxGetApp().get_tab(Preset::TYPE_PRINT)->reload_config();
```

Comment:
These methods are a bit of a mystery to me. These changes work, but I suspect I only need `changed_objects` and not the following two methods. Any advice is appreciated!

Reply:
yeah, we don't need the last two function calls as we don't change config settings here.